File: scripts_php/functions_storage.php
<?php

/** 
 * The following functions test the quota memory and quota storage
 */ 

// Test quota memory by generating a 5 MB string
function func_quota_memory_5_MB($event, $context) 
{
    return func_quota_memory_MB($event, $context, 5);
}

// Test quota memory by generating a 50 MB string
function func_quota_memory_50_MB($event, $context) 
{
    return func_quota_memory_MB($event, $context, 50);
}

// Test quota memory by generating a 90 MB string
function func_quota_memory_90_MB($event, $context) 
{
    return func_quota_memory_MB($event, $context, 90);
}

// Test quota memory by generating a string
function func_quota_memory_MB($event, $context, $iterations)
{
    $message = 'A';
	$iterations = $iterations * 1024 * 1024;
	
    for ($counter = 0; $counter <= $iterations; $counter++) {
        $message .= chr(rand(65,90)); // Random upper char
    }

    echo 'size: ' . strlen($message);

    return [
        'status' => 200,
        'data' => 'OK',
    ];
}

// Test quota storage by generating a 5 MB file
function func_quota_storage_5_MB($event, $context) 
{
    return func_quota_storage_MB($event, $context, 5);
}

// Test quota storage by generating a 50 MB file
function func_quota_storage_50_MB($event, $context) 
{
    return func_quota_storage_MB($event, $context, 50);
}

// Test quota storage by generating a 90 MB file
function func_quota_storage_90_MB($event, $context) 
{
    return func_quota_storage_MB($event, $context, 90);
}

// Test quota storage by generating a file
function func_quota_storage_MB($event, $context, $iterations)
{
	$iterations = $iterations * 1024 * 1024;
    $content = 'This is the new file';
    $this_directory = dirname(dirname(__FILE__));
    $fp = fopen($this_directory . '/../../file.txt', 'a');

    if ($fp) {
        // Open file
        fwrite($fp, $content);
        // Append a string to the file (x times)
        for ($counter = 0; $counter <= $iterations; $counter++) {
            fwrite($fp, 'Append this string to an existing file');
        }
        fclose($fp);
    }

    // Print file size
    echo '/../../file.txt' . ': ' . filesize('/../../file.txt') . ' Byte';

    return [
        'status' => 200,
        'data' => 'OK',
    ];
}

File: scripts_go/functions_storage.go
/*
 * The following functions are testing the chosen storage settings (quota memory, quota storage)
 */

package main

import (
	"fmt"
	"log"
	"math/rand"
	"os"

	"github.com/anexia-it/go-e5e"
)

type entrypoints struct{}

// Test RAM (quota memory) by generating a 5 MB string
func (f *entrypoints) Func_quota_memory_5_MB(event e5e.Event, context e5e.Context) (*e5e.Return, error) {
	Random_string(5243000)

	return &e5e.Return{
		Data: "Hello World",
	}, nil
}

// Test RAM (quota memory) by generating a 50 MB string
func (f *entrypoints) Func_quota_memory_50_MB(event e5e.Event, context e5e.Context) (*e5e.Return, error) {
	Random_string(52430000)

	return &e5e.Return{
		Data: "Hello World",
	}, nil
}

// Test RAM (quota memory) by generating a 500 MB string
func (f *entrypoints) Func_quota_memory_500_MB(event e5e.Event, context e5e.Context) (*e5e.Return, error) {
	Random_string(52430000)
	
	return &e5e.Return{
		Data: "Hello World",
	}, nil
}

// A function which returns a random string with a given length
func Random_string(n int) string {
	var letter = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")

	b := make([]rune, n)
	for i := range b {
		b[i] = letter[rand.Intn(len(letter))]
	}
	fmt.Println(len(string(b)))

	return string(b)
}

// Create 5 MB file to test storage quota (5.1 MB)
func (f *entrypoints) Func_quota_storage_5_MB(event e5e.Event, context e5e.Context) (*e5e.Return, error) {
	create_file(500000)

	return &e5e.Return{
		Data: "Hello World!",
	}, nil
}

// Create 50 MB file to test storage quota (50.11 MB)
func (f *entrypoints) Func_quota_storage_50_MB(event e5e.Event, context e5e.Context) (*e5e.Return, error) {
	create_file(1695000)

	return &e5e.Return{
		Data: "Hello World!",
	}, nil
}

// Create file one folder above "func_code" (read and write)
func (f *entrypoints) Func_quota_storage_100_MB(event e5e.Event, context e5e.Context) (*e5e.Return, error) {
	create_file(3400000)

	return &e5e.Return{
		Data: "Hello World!",
	}, nil
}

func create_file(filesize int){
	// If the file doesn't exist, create it, or append to the file
	fi, err2 := os.OpenFile("../../text.txt", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err2 != nil {
		log.Println(err2)
	}
	defer fi.Close()
	for i := 0; i < filesize; i++ {
		if _, err2 := fi.WriteString("This text appends to the file.\n"); err2 != nil {
			log.Println(err2)
		}
	}
	fi2, err := os.Stat("../../text.txt")
	if err != nil {
		log.Println(err2)
	}
	fmt.Printf("The file is %d bytes long", fi2.Size())
}

func main() {
	if err := e5e.Start(&entrypoints{}); err != nil {
		panic(err)
	}
}
